dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/controllers/WelcomeController.java

```diff
@@ -1,37 +1,165 @@
-	@SuppressWarnings("unused") @FXML
-	private ListView list;
-	@SuppressWarnings("unused") @FXML
-	private void startButtonAction(ActionEvent e) {
+	private DBLoadService dbload;
+	private DirectoryChooser dirChooser;
+	private ProgressDialog progressDialog;
+	private static final String DEFAULT_DB_PATH = "target/db";
+	@SuppressWarnings("unused") @FXML private Button startButton;
+	@SuppressWarnings("unused") @FXML private Button deleteButton;
+	@SuppressWarnings("unused") @FXML private Button loadButton;
+	@SuppressWarnings("unused") @FXML private ListView<String> list;
+	
+	@SuppressWarnings("unused") @FXML 
+	private void startButtonAction(ActionEvent a) {
 		done.setValue(true);
 	}
+	
+	@SuppressWarnings("unused") @FXML 
+	private void deleteButtonAction(ActionEvent a) {
+		try {
+			FileUtils.deleteRecursively(new File(getDBPath()));
+			items.remove(getDBPath());
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+	
+	@SuppressWarnings("unused") @FXML
+	private void loadButtonAction(ActionEvent e) {
+		progressDialog = new ProgressDialog(list.getParent());
+		if (getDBPath() == SELECT_OPTION && selectDirectory() == null) {
+			return;
+		}
+		dbload.restart();
+	}
 
 	@SuppressWarnings("unused") @FXML
-	private void initialize() {
-		ObservableList<String> items = FXCollections.observableArrayList(
-				"nl/tudelft/DNAinator/db/10_strains",
-				"nl/tudelft/DNAinator/db/38_strains",
-				"nl/tudelft/DNAinator/db/100_strains");
+	private void onMouseClicked(MouseEvent e) {
+		if (e.getClickCount() == 2) {
+			if (getDBPath() == SELECT_OPTION) {
+				File dir = selectDirectory();
+				if (dir == null) {
+					return;
+				}
+				dbload.setDatabase(dir.getAbsolutePath());
+				items.add(dir.getAbsolutePath());
+				list.getSelectionModel().select(getDBPath());
+			} else {
+				progressDialog = new ProgressDialog(list.getParent());
+				dbload.restart();
+			}
+		}
+	}
+	
+	@SuppressWarnings("unused") @FXML
+	private void initialize() throws IOException {
+		dirChooser = new DirectoryChooser();
+		dbload = new DBLoadService();
+		
+		dbload.setOnFailed(e -> {
+			new ExceptionDialog(list.getParent(), dbload.getException(),
+					"Database is already in use, please choose another.");
+		});
+		dbload.setOnRunning(e -> progressDialog.show());
+		dbload.setOnSucceeded(e -> {
+			dbProperty.setValue(dbload.getValue());
+			startButton.setDisable(false);
+			progressDialog.close();
+		});
+
+		dbProperty = new SimpleObjectProperty<>(this, "graph");
+		items = FXCollections.observableArrayList(SELECT_OPTION);
+		scanDirectory();
 		list.setItems(items);
+		list.getSelectionModel().select(getDBPath());
+		list.getSelectionModel().selectedItemProperty().addListener((obj, oldV, newV) -> {
+			deleteButton.setDisable(true);
+			if (newV != SELECT_OPTION) {
+				deleteButton.setDisable(false);
+			}
+			dbload.setDatabase(newV);
+			System.out.println(this.getDBPath());
+		});
+	}
+	
+	private String getDBPath() {
```

Comment:
Getter is only used inside the welcome screen and will not be used outside it (method is only used to make code look cleaner).